tensorflow/python/kernel_tests/init_ops_test.py

```diff
@@ -362,6 +362,74 @@ def testInvalidDataType(self):
         dtype=dtypes.string)
 
 
+class VarianceScalingInitializationTest(test.TestCase):
+
+  def testInitializerIdentical(self):
+    for dtype in [dtypes.float32, dtypes.float64]:
+      init1 = init_ops.variance_scaling_initializer(seed=1, dtype=dtype)
+      init2 = init_ops.variance_scaling_initializer(seed=1, dtype=dtype)
+      self.assertTrue(identicaltest(self, init1, init2))
+      init3 = init_ops.variance_scaling_initializer(
+          1.5, seed=1, dtype=dtype)
+      init4 = init_ops.variance_scaling_initializer(
+          1.5, seed=1, dtype=dtype)
+      self.assertTrue(identicaltest(self, init3, init4))
+
+  def testInitializerDifferent(self):
+    for dtype in [dtypes.float32, dtypes.float64]:
+      init1 = init_ops.variance_scaling_initializer(seed=1, dtype=dtype)
+      init2 = init_ops.variance_scaling_initializer(seed=2, dtype=dtype)
+      init3 = init_ops.variance_scaling_initializer(
+          1.5, seed=1, dtype=dtype)
+      self.assertFalse(identicaltest(self, init1, init2))
+      self.assertFalse(identicaltest(self, init1, init3))
+      self.assertFalse(identicaltest(self, init2, init3))
+
+  def testZeroSize(self):
+    shape = [0, 2]
+    with self.test_session():
+      x = variable_scope.get_variable(
+          "x",
+          shape=shape,
+          initializer=init_ops.variance_scaling_initializer())
+      x.initializer.run()
+      self.assertAllEqual(shape, x.eval().shape)
+
+  def testDuplicatedInitializer(self):
+    init = init_ops.variance_scaling_initializer()
+    self.assertFalse(duplicated_initializer(self, init, 1))
+
+  def testInvalidDataType(self):
+    self.assertRaises(
+        ValueError,
+        init_ops.variance_scaling_initializer,
+        dtype=dtypes.string)
+
+  def testNormalDistribution(self):
+    shape = [100, 100]
+    expect_mean = 0.
+    expect_var = 1. / shape[0]
+    init = init_ops.variance_scaling_initializer(distribution='normal')
+    with self.test_session(use_gpu=True):
+      x = variable_scope.get_variable("x", shape=shape, initializer=init)
+      x.initializer.run()
+      x_val = x.eval()
+      self.assertNear(np.mean(x_val), expect_mean, 1e-2)
+      self.assertNear(np.var(x_val), expect_var, 1e-2)
+
+  def testUniformDistribution(self):
+    shape = [100, 100]
+    expect_mean = 0.
+    expect_var = 1. / shape[0]
+    init = init_ops.variance_scaling_initializer(distribution='uniform')
+    with self.test_session(use_gpu=True):
+      x = variable_scope.get_variable("x", shape=shape, initializer=init)
```

Comment:
In these unit tests, I don't think we should use `variable_scope` to create variables (it adds a lot of indirection and isn't future-proof). Prefer using instead:

```python
value = init(...)(shape)
x = variables.Variable(value)
```